bootable-jar/WFLY-13305_Bootable_Jar.adoc

```diff
@@ -0,0 +1,505 @@
+
+===== User application
+
+The WAR or JAR main artifact the Maven project is building is deployed and zipped along with the server. 
+The plugin option context-root=true|false (true by default) allows to rename the war file to ROOT.war 
+(doesn’t apply to other packaging types). 
+
+NB: Application deployment doesn't rely on the deployment scanner. The scanner is not required, it is even suggested to exclude it, useless in a bootable JAR context.
+For an example of how to exclude the _deployment-scanner_ Galleon layer, you can check https://github.com/wildfly-extras/wildfly-jar-maven-plugin/blob/master/examples/jaxrs/pom.xml#L35[this example]
+
+In case the bootable is an “hollow JAR”, no deployment is present in the JAR.
+
+===== Hollow JAR
+
+No WAR/JAR/EAR file is copied into the server deployments. This is controlled by a plugin option.
+The hollow JAR doesn’t require the deployment-scanner to be provisioned in order for the deployment to be taken into account by the runtime.
+At runtime, if the option ```--deployment=<deployment>``` is passed, the deployment artifact is copied to content dir and the xml configuration is updated with the deployment. 
+As an alternative, WildFly CLI can be used to deploy an application inside an hollow jar.
+ 
+
+==== Execute the bootable JAR
+
+This is done thanks to the command: 
+
+```
+java -jar <JAR name>.jar [arguments]
+```
+
+The detailed execution steps are:
+
+* Unzip the server to a directory
+* Handle arguments
+* Start the server
+* Wait for server end 
+
+Alternatively the Maven plugin can be used to run/shutdown the server from Maven wildfly-jar:run|start|shutdown.
+
+==== Configure the Bootable JAR at runtime
+
+The bootable JAR fully relies on the server configuration capabilities. It is not expected to define a new way to configure the server.
+The arguments passed to the bootable JAR are composed of bootable specific arguments and server arguments.
+
+Calling  java <JVM and system properties> -jar <bootable JAR name>.jar --help dumps the available arguments.
+
+==== Log manager
+
+The bootable JAR depends on JBoss log manager. Logging is configured thanks to the logging subsystem.
+
+==== SecurityManager
+
+The bootable JAR can activate the WildFlySecurityManager for the running application server if the `-secmgr` command line argument is 
+added when starting the server.
+
+Permissions to be assigned to deployments should either be added to the `security-manager` subsystem or to a `META-INF/permissions.xml` 
+within the deployment with the latter being preferred.  For either of these approaches to be available the server must be provisioned with the 
+`security-manager` layer present.
+
+
+==== Security
+
+Examples and documentation should be using Elytron security where required and not 
+the legacy security subsystem or legacy security realms as both of these are deprecated for removal in a future release.
```

Comment:
This is important information, so what happens when the legacy security subsystem is used instead?

Reply:
If they are used there should not be an issue, they are existing resources / an existing subsystem after all.  

The use of our MicroProfile JWT integration is already supported exclusively on Elytron only, so this is really a continuation from that point.

The main point of my comment is in relation to us adding a completely new feature to WildFly so at this point we should no longer be providing documentation / examples advising the use of legacy resources which will be removed at some point in the future and instead we should be advocating the latest resources that are not deprecated.

This probably does open up a separate discussion we may need with @bstansberry about what level of support there should be for the legacy security subsystem.  Provisioning the feature pack without specifying any layers I believe would mean they do get the legacy security subsystem.  On the other hand I think we need to check that none of the individual layers have a dependency which brings in the legacy security subsystem.

There are many modules that still have dependencies on PicketBox so the modules may still come in as that is an ongoing task to clean up so it is the presence of the subsystem in the configuration I am really referring to above not the presence of the modules.

Reply:
Thanks @darranl. It seems to me there's ongoing discussion about some modules requiring PicketBox.
That said I am fine with your comment and we basically can let users use legacy security when dealing with layers combinations.